src/components/Identity/index.js

```diff
@@ -100,6 +148,12 @@ const createIdentity = ({ config, logger, cookie }) => {
     commands: {
       getEcid() {
         return optIn.whenOptedIn().then(getEcid);
+      },
+      setCustomerIds(options) {
+        return optIn.whenOptedIn().then(setCustomerIds(options));
```

Comment:
Shouldn't this be:

```
return optIn.whenOptedIn().then(() => setCustomerIds(options));
```

I'm assuming you're wanting to wait for opt in before `setCustomerIds` is called.